subsys/metrics/src/main/java/org/commonjava/indy/metrics/zabbix/cache/ZabbixInit.java

```diff
@@ -0,0 +1,168 @@
+package org.commonjava.indy.metrics.zabbix.cache;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.commonjava.indy.action.IndyLifecycleException;
+import org.commonjava.indy.action.StartupAction;
+import org.commonjava.indy.metrics.zabbix.cache.event.MetricsZabbixCacheEvent;
+import org.commonjava.indy.subsys.datafile.DataFile;
+import org.commonjava.indy.subsys.datafile.DataFileManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Created by xiabai on 5/5/17.
+ */
+@ApplicationScoped
+public class ZabbixInit
+                implements StartupAction
```

Comment:
I'd make this a storage manager with methods that MetricsZabbixCache can reach. I'd remove the @Observes parameters and rename the methods to storeXXX(Map<..>) instead so the MetricsZabbixCache can call them when values change and a particular map needs to be written to disk. I'd make methods for loading each individual map from disk so MetricsZabbixCache can call them. Then, I've remove the interface from this class and put it in MetricsZabbixCache, with the start method calling the load methods in this class.

I also would remove the @Inject MetricsZabbixCache, and let it call this class to load individual Map instances as needed. This class would become almost just a utility class, except for the DataFileManager field injection. It would be responsible for managing the process of writing / reading cache files, and MetricsZabbixCache wouldn't need to know what location was used.